hedera-spring/src/main/java/com/openelements/hedera/spring/implementation/MirrorNodeClientImpl.java

```diff
@@ -65,19 +65,38 @@ public MirrorNodeClientImpl(@NonNull final String mirrorNodeEndpoint) {
     }
 
     @Override
-    public List<Nft> queryNftsByAccount(@NonNull final AccountId accountId) throws HederaException {
+    public Page<Nft> queryNftsByAccount(@NonNull final AccountId accountId) throws HederaException {
         Objects.requireNonNull(accountId, "newAccountId must not be null");
-        final JsonNode jsonNode = doGetCall("/api/v1/accounts/" + accountId + "/nfts");
-        return jsonNodeToNftList(jsonNode);
+        final URI uri = URI.create(
```

Comment:
Can you please update your branch with current `main` branch. I did some updates on the implementation that makes `uri` handling way more easy.

With that update I assume this method looks like this:

```
final String path = /api/v1/accounts/" + accountId + "/nfts";
        final Function<JsonNode, List<Nft>> dataExtractionFunction = node -> getNfts(node);
        return new RestBasedPage<>(objectMapper, restClient.mutate().clone(), path, dataExtractionFunction);
```

Reply:
i refactored the methods, please let me know if i did something wrong.